GunShipAlpha/src/Game.java

```diff
@@ -45,8 +49,8 @@ private void startGameLoop() {
     }
     private void updateExplosions() {
         gs.forEachEnemy(enemy -> {
-            if (enemy instanceof Tank && enemy.isDead()) {
-                ongoingExplosions.addAll(((Tank) enemy).explode());
+            if (enemy instanceof Mine && enemy.isDead()) {
```

Comment:
instanceof checks should _always_ be suspicious. Instead, make a new function on the superclass "handleDeath" and most of them do nothing, while the mine does the "add explosion particles" logic. see?